packages/builders/src/components/button/Button.ts

```diff
@@ -74,7 +70,7 @@ export class ButtonBuilder extends ComponentBuilder<APIButtonComponent> {
 	 * @param url - The URL to use
 	 */
 	public setURL(url: string) {
```

Comment:
Since there's a `setStyle()` method, user's *could* reuse a single builder for constructing multiple styles of buttons...
So I guess that means we should provide a `clearURL()` method? (same applies to `setSkuId()`)

Reply:
The pattern that I adopted is that there wouldn't be a `clear` method for properties that aren't optional, which is technically the case here.

Now that you mention this, I find it pretty compelling to split up the button class depending on style, since it's a very similar pattern to places where we've gotten rid of `setType`-esque stuff in favor of multiple classes.

If we're making an exception here, I'd still rather not make public clear methods, but rather have `setStyle` wipe out the contradictory fields. Thoughts? 🤔